tests/dtypes_test.py

```diff
@@ -977,6 +985,18 @@ def testFloat8PromotionError(self):
                                   ".*8-bit floats do not support implicit promotion"):
         x + y
 
+  @jax.numpy_dtype_promotion('standard')
+  def testFloat4PromotionError(self):
+    for dtype in fp4_dtypes:
```

Comment:
Let's push this iteration outside the test function body, using `@jtu.sample_product(dtype=fp4_dtypes)`

Reply:
fp4_dtypes could be empty.